opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/NonDominatedFronts.java

```diff
@@ -0,0 +1,192 @@
+			Map<Individual, List<Individual>> dominatedIndividualsMap, int[] dominatingIndividualNumber,
+			Map<Individual, Integer> individual2IndexMap) {
+		List<Individual> nextFront = new ArrayList<Individual>();
+		for (Individual dominant : currentFront) {
+			for (Individual dominated : dominatedIndividualsMap.get(dominant)) {
+				dominatingIndividualNumber[individual2IndexMap.get(dominated)]--;
+				if (dominatingIndividualNumber[individual2IndexMap.get(dominated)] == 0) {
+					nextFront.add(dominated);
+				}
+			}
+		}
+		return nextFront;
+	}
+
+	/**
+	 * Compares all possible {@link Individual} pairs. For each individual, stores
+	 * 1) the number of individuals it is dominated by and 2) the set of individuals
+	 * it dominates.
+	 * 
+	 * @param individuals
+	 *            a collection of individuals
+	 * @param dominatedIndividualsMap
+	 *            A map that is filled during the execution of the method. Each
+	 *            individual is mapped onto the set of individuals that are
+	 *            dominated by this individual.
+	 * @param dominatingIndividualNumber
+	 *            An integer array (initialized with zeros) that is filled during
+	 *            the execution of this method. Each individual is associated with
+	 *            an entry of this array. The integer therein is the number of
+	 *            individuals this individual is dominated by.
+	 * @param individual2IndexMap
+	 *            a map mapping each individual onto its index in the
+	 *            dominatingIndividualNumber - array
+	 */
+	protected void determineDomination(Collection<Individual> individuals,
+			Map<Individual, List<Individual>> dominatedIndividualsMap, int[] dominatingIndividualNumber,
+			Map<Individual, Integer> individual2IndexMap) {
+		List<Individual> individualList = new ArrayList<Individual>(individuals);
+		// compare each individual with each other individual
+		for (int i = 0; i < individualList.size(); i++) {
+			for (int j = i + 1; j < individualList.size(); j++) {
+				Individual p = individualList.get(i);
+				Individual q = individualList.get(j);
+				Objectives po = p.getObjectives();
+				Objectives qo = q.getObjectives();
+				if (po.dominates(qo)) {
+					dominatedIndividualsMap.get(p).add(q);
+					dominatingIndividualNumber[individual2IndexMap.get(q)]++;
+				} else if (qo.dominates(po)) {
+					dominatedIndividualsMap.get(q).add(p);
+					dominatingIndividualNumber[individual2IndexMap.get(p)]++;
+				}
+				// Neither of the two points dominates the other one, so that neither the array
+				// keeping track of the domination number nor the map containing the dominating
+				// individuals has to be adjusted. Nothing is done in this case.
+			}
+		}
+	}
+
+	/**
+	 * Returns the {@link Individual}s with the best values for the individual
+	 * {@link Objective}.
+	 * 
+	 * @return the set of the extreme individuals
+	 */
+	public Set<Individual> getExtremeIndividuals() {
```

Comment:
Does this function return the extreme individuals of each front? Why is it part of "NonDominatedFronts"?

Reply:
Moved the method to the ESamplingSurvivorGeneration, which is currently the only class using it.